src/js/utils/getDecodingInfo.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * Returns decoding info for a given media configuration.
+ *
+ * @param {MediaDecodingConfiguration} configuration Media configuration.
+ * @returns {Promise<MediaCapabilitiesInfo>} Media decoding information.
+ */
+export default async function getDecodingInfo(configuration) {
+  if (navigator.mediaCapabilities) {
+    return navigator.mediaCapabilities.decodingInfo(configuration);
+  }
+
+  // If we can't use the Media Capabilities API, try using `canPlayType`.
+  try {
+    const videoEl = document.createElement('video');
+    const typeString = configuration.audio.contentType || configuration.video.contentType;
+    const canPlay = videoEl.canPlayType(typeString);
+
+    return {
+      supported: canPlay === 'probably' || canPlay === 'maybe',
+      smooth: false,
+      powerEfficient: false,
+    };
+  } catch (error) {
+    // Fail quietly.
```

Comment:
When could this be failed in practise?

Reply:
Originally, the `try – catch` was in place to allow the silently fail in a worker context where it's not possible to use DOM methods to construct a `HTMLVideoElement`. 

The code actually used to be more complex, but ended up simple without the `try – catch` being refactored in to a more sensible check.

In any case, per your advice we're now using the more appropriate `MediaSource.isTypeSupported` method with a more targeted check, so the code doesn't exist anymore.

Updated in bdb1b8d.